Fix dates being initialized with client timezone when using timestamps

AppelBoomHD opened the pull request:
Timestamps in mysql and postgressql are by default save as UTC timestamps.
Javascripts Date initializer assumes that dates are written in the client's timezone, when no explicit timezone is given in the initializer. This PR manually adds the UTC timezone to the initializer string.

Another way to this is (https://stackoverflow.com/a/3075893/13799537):
```typescript
// Split timestamp into [ Y, M, D, h, m, s ]
var t = "2010-06-09 13:12:01".split(/[- :]/);

// Apply each element to the Date function
var d = new Date(Date.UTC(t[0], t[1]-1, t[2], t[3], t[4], t[5]));
return new Date(d);
``` 

In TypeScript this gives a couple type-errors, which is why I chose the more simple one-liner solution.

AppelBoomHD committed:
Fix dates being initialized with client timezone when using timestamps

AppelBoomHD committed:
Merge branch 'main' into main

dankochetov commented:
Honestly, I'm not sure if there are hidden implications of adding this. Could you add an integration test for this behavior?

AppelBoomHD committed:
Merge branch 'main' into main

AppelBoomHD committed:
Added tests for timestamp on mysql and postgress

AppelBoomHD committed:
Revert mapFromDriverValue for postgres timestamp

AppelBoomHD commented:
I'm not very familiar with unit testing, but I think I got it working correctly.
Added a test for mysql and postgres databases to test whether the timestamp received from the database is converted to the correct timezone. The test for mysql fails before my change and passes after applying the change.
I realized postgres automatically converts the timestamp to the correct timezone, so I reverted the change related to postgres timestamps.

AppelBoomHD committed:
Merge branch 'main' into main

dankochetov committed:
Merge branch 'main' into main

dankochetov commented on integration-tests/tests/pg.test.ts:
```diff
@@ -1562,3 +1562,22 @@ test.serial('prefixed table', async (t) => {
 	await db.execute(sql`drop table ${users}`);
 });
+
+test.serial('timestamp timezone', async (t) => {
```
Did you actually run this test? We don't run the integration tests in the pipeline yet, so it's up to the developers to run them locally. This one looks like it won't work, because you're not creating the "users" table, you're just defining it, and I'm pretty sure there's already a different "users" table defined in the tests and present in the test DB.

AppelBoomHD replied:
The test succeeded, but I can see now that it doesn't make any sense. It seems I wasn't paying good attention writing this part 😅. Thanks for pointing it out!
I pushed a fix where a new table is actually created, also with a different name.
Since the test was now failing, I noticed that postgres timestamp without timezones were being mapped to the client's timezone, which I had originally suspected. Hence, I've reverted my revert (commit https://github.com/drizzle-team/drizzle-orm/pull/288/commits/34bd785b53e489f58e12a0f4175a0e702252f8af) which made the test succeed.

dankochetov commented on integration-tests/tests/mysql.test.ts:
```diff
@@ -1263,3 +1263,12 @@ test.serial('prefixed table', async (t) => {
+test.serial('timestamp timezone', async (t) => {
+	const { db } = t.context;
+
+	await db.insert(usersTable).values({ name: 'John' });
```
Could you add a test inserting a Date with a non-default timezone specified? I'm curious if it'll change the date string format returned from the driver on selection.

AppelBoomHD replied:
Good point! I added the test as you specified for both mysql and postgres timestamps. Upon running the tests I found out that timestamps are by default inserted respecting the client's timezone as well, instead of in UTC format. I pushed a fix for this behavior, which passed all the tests.

AppelBoomHD committed:
Revert "Revert mapFromDriverValue for postgres timestamp"

This reverts commit 34bd785b53e489f58e12a0f4175a0e702252f8af.

AppelBoomHD committed:
Fix timestamp tests and include test for dates with non-default timezone

AppelBoomHD committed:
Insert timestamp with correct timezone for non-default insert for pg and mysql

dankochetov reviewed (changes requested):
Seems like you have a code formatter (Prettier, probably) enabled on save, which changed the formatting quite a bit. We use Dprint for this repo, so please re-format the files with it (it might not rollback some formatting you already did, so you'll have to change it back manually).

AppelBoomHD committed:
Revert styling changes from previous commits

dankochetov commented:
@AppelBoomHD please resolve the merge conflicts

AppelBoomHD committed:
Merge branch 'main' into main

AppelBoomHD committed:
Merge branch 'main' into main
